File: 01_Calculator/app/src/main/java/com/example/a01_calculator/MainActivity.java
package com.example.a01_calculator;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    Button b1;
    Button b2;
    Button b3;
    Button b4;
    Button b5;
    Button b6;
    Button b7;
    Button b8;
    Button b9;
    Button b0;
    Button clear;
    Button equal;

    Button plus;
    Button minus;
    Button div;
    Button mul;

    EditText editText;

    public static final String TAG = "cal_tag";
    String t1 = "";
    String sign = "";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        b1 = findViewById(R.id.button);
        b2 = findViewById(R.id.button2);
        b3 = findViewById(R.id.button3);
        b4 = findViewById(R.id.button4);
        b5 = findViewById(R.id.button5);
        b6 = findViewById(R.id.button6);
        b7 = findViewById(R.id.button7);
        b8 = findViewById(R.id.button8);
        b9 = findViewById(R.id.button9);
        b0 = findViewById(R.id.button0);

        clear = findViewById(R.id.button10);
        equal = findViewById(R.id.button11);

        plus = findViewById(R.id.buttonadd);
        minus = findViewById(R.id.buttonminus);
        mul = findViewById(R.id.buttonmultiply);
        div = findViewById(R.id.buttondivide);

        editText = findViewById(R.id.editTextNumber);

//        b1.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//               set
//            }
//        });

            b1.setOnClickListener(this);
            b2.setOnClickListener(this);
        b3.setOnClickListener(this);
        b4.setOnClickListener(this);
        b5.setOnClickListener(this);
        b6.setOnClickListener(this);
        b7.setOnClickListener(this);
        b8.setOnClickListener(this);
        b9.setOnClickListener(this);
        b0.setOnClickListener(this);

        clear.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editText.setText("");
            }
        });
        plus.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                t1 = editText.getText().toString();
                editText.setText("");
                sign = "+";
            }
        });
        minus.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                t1 = editText.getText().toString();
                editText.setText("");
                sign = "-";
            }
        });
        mul.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                t1 = editText.getText().toString();
                editText.setText("");
                sign = "*";
            }
        });
        div.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                t1 = editText.getText().toString();
                editText.setText("");
                sign = "/";
            }
        });
    }

    protected void setText(String v) {
        editText.setText(editText.getText().toString() + v);
    }

    public void showResult(View v) {
        if( sign.isEmpty()) {
            return;
        }
        String value = editText.getText().toString();
        int a = Integer.valueOf(t1);
        int b = Integer.valueOf(value);
        int ans = 0;
        if(sign == "+") {
            ans = a + b;
        } else if( sign == "-") {
            ans = a - b;
        } else if (sign == "*") {
            ans = a * b;
        } else if (sign == "/") {
            ans = a / b;
        }
        editText.setText(Integer.toString(ans));
        sign = "";
    }

    @Override
    public void onClick(View v) {
        if (v == b1) {
            setText("1");
        } else if (v == b2) {
            setText("2");
        } else if (v == b2) {
            setText("2");
        } else if (v == b3) {
            setText("3");
        } else if (v == b4) {
            setText("4");
        } else if (v == b5) {
            setText("5");
        } else if (v == b6) {
            setText("6");
        } else if (v == b7) {
            setText("7");
        } else if (v == b8) {
            setText("8");
        } else if (v == b9) {
            setText("9");
        } else if (v == b0) {
            setText("0");
        }
    }
}

File: 04_TicTacToe/app/src/main/java/com/example/tictactoe/MainActivity.kt
package com.example.tictactoe

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), View.OnClickListener {

    var boardStatus = Array(3) { IntArray(3)}
    lateinit var btns: Array<Array<Button>>

    var PLAYER = true
    var turnCount = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btns = arrayOf(
            arrayOf(button1, button2, button3),
            arrayOf(button4, button5, button6),
            arrayOf(button7, button8, button9)
        )

        for(i in btns) {
            for(btn in i) {
                btn.setOnClickListener(this)
            }
        }

        initializeBoardStatus()

        ResetBtn.setOnClickListener {
            PLAYER = true
            turnCount = 0
            initializeBoardStatus()
        }
    }

    private fun initializeBoardStatus() {
        for(i in 0..2) {
            for(j in 0..2) {
                boardStatus[i][j] = -1
                btns[i][j].apply {
                    text = ""
                    isEnabled = true
                }
            }
        }
        displayPlayer.setText("Player ${if(PLAYER) "X" else "0"} TURN")

    }

    override fun onClick(view: View) {
        when(view.id) {
            R.id.button1 -> {
                updateValue(row = 0, col = 0, player = PLAYER)
            }
            R.id.button2 -> {
                updateValue(row = 0, col = 1, player = PLAYER)
            }
            R.id.button3 -> {
                updateValue(row = 0, col = 2, player = PLAYER)
            }
            R.id.button4 -> {
                updateValue(row = 1, col = 0, player = PLAYER)
            }
            R.id.button5 -> {
                updateValue(row = 1, col = 1, player = PLAYER)
            }
            R.id.button6 -> {
                updateValue(row = 1, col = 2, player = PLAYER)
            }
            R.id.button7 -> {
                updateValue(row = 2, col = 0, player = PLAYER)
            }
            R.id.button8 -> {
                updateValue(row = 2, col = 1, player = PLAYER)
            }
            R.id.button9 -> {
                updateValue(row = 2, col = 2, player = PLAYER)
            }
        }
    }

    private fun updateValue(row: Int, col: Int, player: Boolean) {
        btns[row][col].apply {
            isEnabled = false
            setText(if(player) "X" else "0")
        }
        boardStatus[row][col] = if(player) 1 else 0
        PLAYER = !PLAYER
        displayPlayer.setText("Player ${if(!player) "X" else "0"} TURN")

        if (checkWinner()) {
            for(i in 0..2) {
                for(j in 0..2) {
                    btns[i][j].apply {
                        isEnabled = false
                    }
                }
            }
        } else {
            turnCount++
            if(turnCount == 9) {
                displayPlayer.setText("GAME OVER")
            }
        }
    }

    private fun checkWinner(): Boolean {
        if(boardStatus[0][0] == boardStatus[1][1] && boardStatus[0][0] == boardStatus[2][2]) {
            if(boardStatus[0][0] != -1) {
                displayPlayer.setText("Player ${if(boardStatus[0][0] == 1) "X" else "0"} WINNER")
                return true
            }
        }
        if(boardStatus[0][2] == boardStatus[1][1] && boardStatus[0][2] == boardStatus[2][0]) {
            if(boardStatus[0][2] != -1) {
                displayPlayer.setText("Player ${if(boardStatus[0][0] == 1) "X" else "0"} WINNER")
                return true
            }
        }
        for(i in 0..2) {
            if (boardStatus[i][0] == boardStatus[i][1] && boardStatus[i][0] == boardStatus[i][2]) {
                if(boardStatus[i][0] != -1) {
                    displayPlayer.setText("Player ${if(boardStatus[i][0] == 1) "X" else "0"} WINNER")
                    return true
                }
            }
            if(boardStatus[0][i] == boardStatus[1][i] && boardStatus[0][i] == boardStatus[2][i]) {
                if(boardStatus[0][i] != -1) {
                    displayPlayer.setText("Player ${if(boardStatus[0][i] == 1) "X" else "0"} WINNER")
                    return true
                }
            }
        }
        return false
    }
}